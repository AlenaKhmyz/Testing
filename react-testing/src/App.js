import logo from './logo.svg';
import './App.css';
import { useState, useEffect } from 'react'

function App() {
  const [data, setData] = useState(null)
  const [toggle, setToggle] = useState(false)

  const onClick = () => setToggle(prev => !prev)

  useEffect(() => {
    setTimeout(() => {
      setData({})
    }, 100)
  }, [])

  return (
    <div className="App">
      {toggle === true && <div data-testid='toggle-elem'>toggle</div>}
      {data && <div style={{ color: 'blue' }}>data</div>}
      <h1>hi</h1>
      <button onClick={onClick} data-testid='toggle-btn'>click me!</button>
      <input type='text' placeholder='input value' />
    </div>
  );
}

export default App;
