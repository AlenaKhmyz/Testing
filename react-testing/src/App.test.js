import { render, screen, fireEvent, queryByTestId } from '@testing-library/react';
import App from './App';

describe ('TEST API', () => {
  test('renders learn react link', () => {
    render(<App />); //передается компонент, который хотим протестировать
    const hiElement = screen.getByText(/hi/i);//здесь получаем при помощи объекта screen какой-то элемент
    const btnElement = screen.getByRole('button');
    const inputElement = screen.getByPlaceholderText(/input value/i);
    expect(hiElement).toBeInTheDocument();
    expect(btnElement).toBeInTheDocument();
    expect(inputElement).toMatchSnapshot();
    screen.debug()//помогает с поиском ошибки
  });

  test('renders learn react link', async () => {
    render(<App />);
    screen.debug()
    const hiElement = await screen.findByText(/data/i);
    expect(hiElement).toBeInTheDocument();
    expect(hiElement).toHaveStyle({color: 'blue'})
    screen.debug()
  });

   test('click event', () => {
    render(<App />);
     const btn = screen.getByTestId('toggle-btn');
     expect(screen.queryByTestId('toggle-elem')).toBeNull()
     fireEvent.click(btn)//предназначен для работы с событиями
     expect(screen.queryByTestId('toggle-elem')).toBeInTheDocument()
     fireEvent.click(btn)
     expect(screen.queryByTestId('toggle-elem')).toBeNull()
  });
})


