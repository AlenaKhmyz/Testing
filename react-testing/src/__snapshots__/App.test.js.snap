// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TEST API renders learn react link 1`] = `
<input
  placeholder="input value"
  type="text"
/>
`;

exports[`TEST API renders learn react link 2`] = `
<h1>
  hi
</h1>
`;

exports[`renders learn react link 1`] = `
<input
  placeholder="input value"
  type="text"
/>
`;
